src/core.js

```diff
@@ -1300,12 +1300,12 @@ $.extend( $.validator, {
 			this.settings.messages[ element.name ].remote = previous.message;
 
 			param = typeof param === "string" && { url: param } || param;
-
-			if ( previous.old === value ) {
+			optionDataString = JSON.stringify( $.param( $.extend( { data: value }, param.data ) ) );
```

Comment:
The $.param serializes the object into a query string, so that it can be stored as a property on the element itself (In the same way the Ajax request would send it to the api endpoint). This guarantees that jquery val will recognize when something about the request has changed instead of just the value of the validating field.